bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -82,6 +82,10 @@ data:
         export KAFKA_CFG_BROKER_ID="$((ID + {{ .Values.minBrokerId }}))"
     fi
 
+    {{- if eq .Values.brokerRackAssignment "aws-az" }}
+    export KAFKA_CFG_BROKER_RACK=$(curl "http://169.254.169.254/latest/meta-data/placement/availability-zone-id")
```

Comment:
Hi,
  The current image `bitnami/spark:3.3.1-debian-11-r1` does not contains `curl` program.
  Also, I don't like the idea of having a hardcoded IP. I'd rather to have a hostname.

Reply:
Hi, thanks for your review.
1. This chart has `Image:         docker.io/bitnami/kafka:3.3.1-debian-11-r1` not spark. https://github.com/bitnami/charts/blob/ad525e69bbe52087ab312a046f41fb99e2cddb07/bitnami/kafka/values.yaml#L70-L72
2. this image has `curl` https://github.com/bitnami/containers/blob/main/bitnami/kafka/3.3/debian-11/Dockerfile#L22

3. In AWS, EC2 can get placement information from 169.254.169.254. [document here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html), And there was no DNS with AWS metadata IP. hardcorded IP was necessary in this case.

Reply:
Hi,
  I am sorry, I was dealing with an spark issue and I mixed things.

Reply:
LGTM, let's use the IP meanwhile.

Reply:
@DingGGu Regarding this, why don't get the zone from the kubernetes zone label via the kubernetes API, so using  https://kubernetes.default.svc/api/v1/nodes/${NODE_NAME} and extract it from the `topology.kubernetes.io/zone` label.
This approach could be cloud vendor generic.
What do you think?

Reply:
Yes, I already know this mechanism, and I've checked that other Charts are borrowing it. However, in order to get the node label, the pod will need to be granted Kubernetes permission via ServiceAccount.
Normally, non-"Cluster Administrator" that, regular developers will not have Role and RoleBinding privileges. I wanted to prevent regular developers just can deploy Kafka via the Bitnami Helm Chart.

Therefore, I tried to option for brokerAssignment in string format so that various types can be supported in future.

Reply:
@rubenvw-ngdata Yes, I already know this mechanism, and I've checked that other Charts are borrowing it. However, in order to get the node label, the pod will need to be granted Kubernetes permission via ServiceAccount.
Normally, non-"Cluster Administrator" that, regular developers will not have Role and RoleBinding privileges. I wanted to prevent regular developers just can deploy Kafka via the Bitnami Helm Chart.

Therefore, I tried to option for brokerAssignment in string format so that various types can be supported in future.